projects/rero/ng-core/src/lib/record/files/file/file.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { RecordUiService } from '../../record-ui.service';
+import { of } from 'rxjs';
+import { ActionStatus } from '../../action-status';
+
+@Component({
+  selector: 'ng-core-record-file',
+  templateUrl: './file.component.html',
+  styleUrls: ['./file.component.css'],
+})
+export class FileComponent implements OnInit {
+  @Input()
+  record: any;
+
+  @Input()
+  file: any;
+
+  @Input()
+  type: any;
+
+  @Input()
+  hasChildren: boolean;
+
+  @Input()
+  infoExcludedFields: Array<string> = [];
+
+  @Output() deleteFile: EventEmitter<void> = new EventEmitter();
+  @Output() manageFile: EventEmitter<void> = new EventEmitter();
+  @Output() editMetadataFile: EventEmitter<void> = new EventEmitter();
+
+  canUpdateMetadata: ActionStatus;
+  canUpdate: ActionStatus;
+  canDelete: ActionStatus;
+
+  constructor(private _recordUiService: RecordUiService) {}
+
+  ngOnInit(): void {
+    this.loadPermission();
+  }
+
+  // Check if a file can be added.
+  loadPermission() {
+    const config = this._recordUiService.getResourceConfig(this.type);
+    let obs$ = config.files.canUpdate
+      ? config.files.canUpdate(this.record, this.file)
+      : of({ can: false, message: '' });
+    obs$.subscribe((result: ActionStatus) => {
+      this.canUpdate = result;
+    });
+    obs$ = config.files.canDelete ? config.files.canDelete(this.record, this.file) : of({ can: false, message: '' });
```

Comment:
I didn't know that it was possible to reassigned a new observable to a subscribed observable.